src/main/asciidoc/transactions.adoc

```diff
@@ -33,82 +41,257 @@ static class Config extends AbstractCouchbaseConfiguration {
+is and whether to include package information (which isn't mandatory). For example,
+"Exception" will match nearly anything and will probably hide other
+rules. "java.lang.Exception" would be correct if "Exception"
+were meant to define a rule for all checked exceptions. With more unique
+exception names such as "BaseBusinessException" there is likely no
+need to use the fully qualified class name for the exception pattern. Furthermore,
+rollback rules defined via patterns may result in unintentional matches for
+similarly named exceptions and nested classes. This is due to the fact that a
+thrown exception is considered to be a match for a given pattern-based rollback
+rule if the name of thrown exception contains the exception pattern configured
+for the rollback rule. For example, given a rule configured to match against
+"com.example.CustomException", that rule will match against an exception
+named com.example.CustomExceptionV2 (an exception in the same package as
+CustomException but with an additional suffix) or an exception named
+com.example.CustomException$AnotherException (an exception declared as
+a nested class in CustomException).
+ 
+For specific information about the semantics of other attributes in this
+annotation, consult the org.springframework.transaction.TransactionDefinition
+and org.springframework.transaction.interceptor.TransactionAttribute javadocs.
+ 
+=== Transaction Management
+ 
+This annotation commonly works with thread-bound transactions managed by a
+org.springframework.transaction.PlatformTransactionManager, exposing a
+transaction to all data access operations within the current execution thread.
+Note: This does NOT propagate to newly started threads within the method.
+
+Couchbase does not supply a reactive transaction manager, therefore it cannot support
+@Transactional on reactive methods. Couchbase provides reactive transaction support
+via the CouchbaseTransactionalOperator. 
 
 .Transaction Access
 ====
 [source,java]
 ----
-@Autowired
-Transactions transactions;
+// this object must be provided by the Spring framework. By @Autowire or from ac.getBean().
+// If created directly with it's constructor, the method interceptor will not be called.
+@Autowired PersonService personService;
+
+    Person p = personService.savePerson(WalterWhite);
+
 
-@Autowired
-CouchbaseClientFactory couchbaseClientFactory;
+@Service
+@Component
+public class PersonService {
+
+    final CouchbaseOperations personOperations;
+
+    public PersonService(CouchbaseOperations ops) {
+        personOperations = ops;
+    }
+
+    @Transactional(transactionManager = BeanNames.COUCHBASE_TRANSACTION_MANAGER) // default, shown for clarity
+    public Person replacePerson(String id) {
+        Person p = personOperations.findById(Person.class).one(id);
+        p.setFirstname(p.getFirstname()+" Junior");
+        return personOperations.replaceById(Person.class).one(p);
+    }
```

Comment:
Other things we probably want to mention in relation to @Transactional:

- Which operations are supported.
- **Very important**: that this is going to get retried.  They want to fetch their documents inside it, to avoid CAS problems.  They don't want any side-effects (apart from transaction operations of course).
- Do Spring users usually use template methods directly like this, or repository ones?  Maybe want examples for both, leading with the most popular method.

Reply:
The middle one is now done, though it visually seems a bit buried to me in the midst of the initial example.

The other two aren't AFAICT, but they are less critical.

Reply:
Aw crap. I had mentioned how "Warning:" made a nice box, but then I just stuck it in the code example.

Reply:
Heh